radix-engine/src/system/system_modules/transaction_runtime/module.rs

```diff
@@ -53,14 +69,21 @@ impl TransactionRuntimeModule {
         self,
         is_success: bool,
     ) -> (Vec<(EventTypeIdentifier, Vec<u8>)>, Vec<(Level, String)>) {
-        if !is_success {
-            return (Vec::new(), self.logs);
-        }
+        let mut results = Vec::new();
+
+        for Event {
+            mut type_identifier,
+            payload,
+            revert_on_failure,
+        } in self.events.into_iter()
+        {
+            // Revert if failure
+            if revert_on_failure && !is_success {
```

Comment:
"revert" seems like wrong word choice since there is no "original" event. Perhaps "discard" is better.